### PR TITLE
Remove Carthage dependency build products

### PR DESCRIPTION
**Pull request by joewalsh**
Labeled as WIP since it's still up for discussion here: https://phabricator.wikimedia.org/T158213

### CODE REVIEW

**Comment by joewalsh** [comment]
jenkins, test this please

**Comment by montehurd** [comment]
Nice!

File: scripts/setup [hunk 1]
```diff
@@ -7,4 +7,8 @@ rbenv install -s
 gem install bundler
 bundle install
 rm -rf .git/hooks/pre-commit
-cp -f scripts/clang_format_git_diff .git/hooks/pre-commit
\ No newline at end of file
+cp -f scripts/clang_format_git_diff .git/hooks/pre-commit
+
+/usr/local/bin/carthage bootstrap --platform iOS
+
+plutil -replace CFBundleVersion -string 4.0.0 Carthage/Build/iOS/SDWebImage.framework/Info.plist
```
Comment (amrox) [inline scripts/setup]: What's this for?

File: Wikipedia.xcodeproj/project.pbxproj [hunk 2]
```diff
@@ -6531,7 +6531,7 @@
 			);
 			runOnlyForDeploymentPostprocessing = 0;
 			shellPath = /bin/sh;
-			shellScript = "if ! cmp -s \"$SRCROOT/Cartfile.resolved\" \"$SRCROOT/Carthage/Cartfile.resolved_at_last_bootstrap\"\nthen\ncarthage bootstrap --platform iOS\nplutil -replace CFBundleVersion -string 4.0.0 \"$SRCROOT/Carthage/Build/iOS/SDWebImage.framework/Info.plist\"\nmkdir -p \"$SRCROOT/Carthage\"\ncp \"$SRCROOT/Cartfile.resolved\" \"$SRCROOT/Carthage/Cartfile.resolved_at_last_bootstrap\"\nfi";
+			shellScript = "if ! cmp -s \"$SRCROOT/Cartfile.resolved\" \"$SRCROOT/Carthage/Cartfile.resolved_at_last_bootstrap\"\nthen\ncarthage bootstrap --platform iOS 2> /dev/null\nplutil -replace CFBundleVersion -string 4.0.0 \"$SRCROOT/Carthage/Build/iOS/SDWebImage.framework/Info.plist\"\nmkdir -p \"$SRCROOT/Carthage\"\ncp \"$SRCROOT/Cartfile.resolved\" \"$SRCROOT/Carthage/Cartfile.resolved_at_last_bootstrap\"\nfi";
```
Comment (amrox) [inline Wikipedia.xcodeproj/project.pbxproj]: Why stderr > /dev/null?

File: scripts/setup (hunk 1 above)
Comment (joewalsh) [inline scripts/setup]: SDWebImage is lacking a CFBundleVersion which causes a failure when trying to submit the app to the store. This fix-up could be done only on archive, but I also figured I'd eventually want to add something to fix the build warnings that the frameworks are currently generating.

File: Wikipedia.xcodeproj/project.pbxproj (hunk 2 above)
Comment (joewalsh) [inline Wikipedia.xcodeproj/project.pbxproj]: Outputs to stderr by the script appear in Xcode as build errors - it then confused Xcode when it got errors but still returned 0. The only true failure should be if carthage doesn't return 0.

**Comment by joewalsh** [comment]
jenkins, test this please

**Comment by joewalsh** [comment]
jenkins, test this please

**Pull request merged**